MetaLamp/lending-pool/src/Plutus/Contracts/Endpoints.hs

```diff
@@ -186,10 +188,87 @@ withdraw aave WithdrawParams {..} = do
 
     State.updateReserve aave wpAsset (reserve { rAmount = rAmount reserve - wpAmount })
 
+data BorrowParams =
+    BorrowParams {
+        bpAsset      :: AssetClass,
+        bpAmount     :: Integer,
+        bpOnBehalfOf :: PubKeyHash
+    }
+    deriving stock    (Show, Generic)
+    deriving anyclass (ToJSON, FromJSON, ToSchema)
+
+PlutusTx.unstableMakeIsData ''BorrowParams
+PlutusTx.makeLift ''BorrowParams
+
+borrow :: (HasBlockchainActions s) => Aave -> BorrowParams -> Contract w s Text ()
+borrow aave BorrowParams {..} = do
+    reserve <- State.findAaveReserve aave bpAsset
+
+    utxos <-
+        Map.filter ((> 0) . flip assetClassValueOf bpAsset . txOutValue . txOutTxOut)
+        <$> utxoAt (Core.aaveAddress aave)
+    let inputs = (\(ref, tx) -> OutputValue ref tx Core.BorrowRedeemer) <$> Map.toList utxos
+    let payment = assetClassValue (rCurrency reserve) bpAmount
+    let remainder = assetClassValue (rCurrency reserve) (rAmount reserve - bpAmount)
```

Comment:
There is seems to be a pattern - when we deduct a value from a script, we have to calculate remainder to return it to the script. I think I'll ask about this on tomorrow's q&a. If this is the correct way to do it, I think we should move this to a separate function(in `TxUtils`), to avoid duplication.